### PR TITLE
feat:#1810

### PR DESCRIPTION
**Pull request by songzhibin97**
The set point was not found

### CODE REVIEW

**Comment by taoyuanyuan** [comment]
> The set point was not found

Maybe we should use variable to set upstreamid that this is not bound to downstreamConn.

**Comment by taoyuanyuan** [comment]
for example
https://github.com/mosn/mosn/blob/d12ff6bb606ea69a2ecfab1327cdcc87a2257d0a/pkg/proxy/var.go#L56

**Comment by songzhibin97** [comment]
Like this?

**Comment by taoyuanyuan** [comment]
> Like this?

yeah~ 
but we should set conid, and maybe we can set it by `activeClient.codecClient.ConnID()`
https://github.com/mosn/mosn/blob/bb9f0b53e466079c55862caba8729aca3ac6964f/pkg/stream/xprotocol/connpool_multiplex.go#L146

File: pkg/stream/xprotocol/connpool_multiplex.go [hunk 1]
```diff
@@ -177,6 +178,8 @@ func (p *poolMultiplex) NewStream(ctx context.Context, receiver types.StreamRece
 		return host, nil, types.Overflow
 	}
 
+	mosnctx.WithValue(ctx, types.ContextUpstreamConnectionID, strconv.FormatUint(activeClient.codecClient.ConnID(), 10))
```
Comment (taoyuanyuan) [inline pkg/stream/xprotocol/connpool_multiplex.go:181]: we should use `variable.SetString` to set id.

File: pkg/proxy/var.go [hunk 2]
```diff
@@ -50,6 +50,7 @@ var (
 		variable.NewStringVariable(types.VarUpstreamHost, nil, upstreamHostGetter, nil, 0),
 		variable.NewStringVariable(types.VarUpstreamTransportFailureReason, nil, upstreamTransportFailureReasonGetter, nil, 0),
 		variable.NewStringVariable(types.VarUpstreamCluster, nil, upstreamClusterGetter, nil, 0),
+		variable.NewStringVariable(types.VarUpstreamConnectID, nil, upstreamConnectId, nil, 0),
```
Comment (taoyuanyuan) [inline pkg/proxy/var.go:53]: we should use `variable.DefaultStringSetter` to set variable, you can refer to `types.VarProxyGlobalTimeout`

**Comment by taoyuanyuan** [comment]
You can learn more details about variable through this doc. https://mosn.io/docs/developer-guide/variable/

**Comment by songzhibin97** [comment]
> You can learn more details about variable through this doc. https://mosn.io/docs/developer-guide/variable/

get

File: pkg/log/proxylog.go [hunk 3]
```diff
@@ -121,6 +122,10 @@ func traceInfo(ctx context.Context) string {
 	if traceId != nil {
 		tid = traceId.(string)
 	}
+	upstreamConnectionId := mosnctx.Get(ctx, types.ContextUpstreamConnectionID) // string
```
Comment (taoyuanyuan) [inline pkg/log/proxylog.go:125]: use `variable.GetString` to replace `mosnctx.Get`.

Comment (songzhibin97) [reply pkg/log/proxylog.go:125]: 使用 variable.GetString 在测试中没办法提前引用注册 proxy.builtinVariables init,看来 types.ContextUpstreamConnectionID应该是有问题

Comment (songzhibin97) [reply pkg/log/proxylog.go:125]: 另外在 Newstream中 mosnctx.WithValue(ctx, types.ContextUpstreamConnectionID, strconv.FormatUint(activeClient.codecClient.ConnID(), 10)) 是否也有问题

Comment (taoyuanyuan) [reply pkg/log/proxylog.go:125]: 我想了下，那就还是先直接用mosnctx.WithValue的方式吧，不然downid和traceid是ctx，然后upid是variable，比较分裂。
后续统一都修改为变量。

File: pkg/stream/xprotocol/connpool_multiplex.go (hunk 1 above)
Comment (taoyuanyuan) [inline pkg/stream/xprotocol/connpool_multiplex.go:181]: add this logic in other connpools

Comment (songzhibin97) [reply pkg/stream/xprotocol/connpool_multiplex.go:181]: Help to check.
types.ContextUpstreamConnectionID
types.VarUpstreamConnectID

File: pkg/proxy/var.go [hunk 4]
```diff
@@ -50,6 +50,7 @@ var (
 		variable.NewStringVariable(types.VarUpstreamHost, nil, upstreamHostGetter, nil, 0),
 		variable.NewStringVariable(types.VarUpstreamTransportFailureReason, nil, upstreamTransportFailureReasonGetter, nil, 0),
 		variable.NewStringVariable(types.VarUpstreamCluster, nil, upstreamClusterGetter, nil, 0),
+		variable.NewStringVariable(types.VarUpstreamConnectID, nil, nil, variable.DefaultStringSetter, 0),
```
Comment (taoyuanyuan) [inline pkg/proxy/var.go:53]: delete this.

**Comment by taoyuanyuan** [comment]
fix conflicting file.

File: pkg/log/proxylog.go (hunk 3 above)
Comment (taoyuanyuan) [inline pkg/log/proxylog.go:125]: using uint64 to keep the consistent style.

File: pkg/log/proxylog_test.go [hunk 5]
```diff
@@ -74,9 +76,12 @@ func TestProxyLog2(t *testing.T) {
 	}
 	traceId := "0abfc19515355177863163255e6d87"
 	connId := uint64(1)
-	proxyMsg := fmt.Sprintf("[%d,%s]", connId, traceId)
+	upstreamConnID := uint64(2)
+	proxyMsg := fmt.Sprintf("[%d,%s,%s]", connId, upstreamConnID, traceId)
```
Comment (taoyuanyuan) [inline pkg/log/proxylog_test.go:80]: error format.

**Review by taoyuanyuan — APPROVED** [review]
LGTM

**Comment by nejisama** [comment]
提供一下新的运行日志看看效果是不是符合预期的？比如运行几个example看一下

**Review by nejisama — APPROVED** [review]

**Pull request merged**